src/aeb_fate_maps.erl

```diff
@@ -124,6 +131,61 @@ write_cache(Key, ?FATE_MAP_TOMBSTONE, Map) ->
 write_cache(Key, Val, Map) ->
     Map#{ Key => Val }.
 
+%% -- Reference counting -----------------------------------------------------
+
+-type refcount() :: #{id() => pos_integer()}.
+
+-spec refcount_zero() -> refcount().
+refcount_zero() -> #{}.
+
+-spec refcount_diff(refcount(), refcount()) -> refcount().
+refcount_diff(New, Old) ->
+    maps:fold(fun(K, N, C) -> maps:update_with(K, fun(M) -> M - N end, -N, C) end,
```

Comment:
So this may return a negative refcount? (While the type suggests otherwise.)

Reply:
Yes indeed. The type should really have `integer()` instead of `pos_integer()`.